container/mock_docker_client.go

```diff
@@ -0,0 +1,89 @@
+// +build ignore
+package container
+
+import (
+	"errors"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+type MockDockerClient struct {
```

Comment:
If you want to use files from another package (the point of this exercise), you can't name them `*_test.go` because they aren't available to be imported. But it doesn't have to be named `*_test.go` to exclude it from the binary build. I added the build tag at the top of the file to exclude it.